src/main/java/com/k3ntako/HTTPServer/Server.java

```diff
@@ -25,23 +25,43 @@ public Server(
 
   public void run() throws IOException {
     var clientSocket = serverSocket.accept();
+    String responseStr = "";
 
-    serverIO.init(clientSocket);
-
-    var request = this.requestHandler.handleRequest(serverIO);
-    var route = this.router.routeRequest(request);
-    var response = route.createResponse();
-    serverIO.sendData(response);
+    try {
+      serverIO.init(clientSocket);
 
-    this.close(clientSocket);
+      var request = this.requestHandler.handleRequest(serverIO);
+      var response = this.router.routeRequest(request);
+      responseStr = response.createResponse();
+    } catch (HTTPError e) {
+      e.printStackTrace();
+      var response = generateErrorResponse(e.getStatus(), e);
+      responseStr = response.createResponse();
+    } catch (Exception e) {
+      e.printStackTrace();
+      var response = generateErrorResponse(500, e);
+      responseStr = response.createResponse();
+    } finally {
+      serverIO.sendData(responseStr);
+      this.close(clientSocket);
+    }
```

Comment:
Maybe this is too high up. Have an abstraction
- Server should just know about the starting the server
- Middleware would be where it should be